src/renderer/components/hotbar/hotbar-icon.tsx

```diff
@@ -0,0 +1,132 @@
+import "./hotbar-icon.scss";
+
+import React, { DOMAttributes } from "react";
+import { observer } from "mobx-react";
+import { cssNames, IClassName } from "../../utils";
+import { Tooltip } from "../tooltip";
+import { Avatar } from "@material-ui/core";
+import { CatalogEntity, CatalogEntityContextMenu, CatalogEntityContextMenuContext } from "../../../common/catalog-entity";
+import { Menu, MenuItem } from "../menu";
+import { Icon } from "../icon";
+import { observable } from "mobx";
+import { navigate } from "../../navigation";
+import { hotbarStore } from "../../../common/hotbar-store";
+import { ConfirmDialog } from "../confirm-dialog";
+
+interface Props extends DOMAttributes<HTMLElement> {
+  entity: CatalogEntity;
+  index: number;
+  className?: IClassName;
+  errorClass?: IClassName;
+  isActive?: boolean;
+}
+
+@observer
+export class HotbarIcon extends React.Component<Props> {
+  @observable.deep private contextMenu: CatalogEntityContextMenuContext;
+  @observable menuOpen = false;
+
+  componentDidMount() {
+    this.contextMenu = {
+      menuItems: [],
+      navigate: (url: string) => navigate(url)
+    };
+  }
+
+  get iconString() {
+    let splittedName = this.props.entity.metadata.name.split(" ");
+
+    if (splittedName.length === 1) {
+      splittedName = splittedName[0].split("-");
+    }
+
+    if (splittedName.length === 1) {
+      splittedName = splittedName[0].split("@");
+    }
+
+    splittedName = splittedName.map((part) => part.replace(/\W/g, ""));
```

Comment:
See #2331 but I really don't think we should be ignoring non-english characters.